File: Ruby/number-of-strings-that-appear-as-substrings-in-word.rb
# https://leetcode.com/problems/number-of-strings-that-appear-as-substrings-in-word/

# @param {String[]} patterns
# @param {String} word
# @return {Integer}
def num_of_strings(patterns, word)
  count = 0
  patterns.each do |pattern|
    count += 1 if word.include?(pattern)
  end
  count
end

File: Python/number-of-strings-that-appear-as-substrings-in-word.py
# https://leetcode.com/problems/number-of-strings-that-appear-as-substrings-in-word/

class Solution:
    def numOfStrings(self, patterns: List[str], word: str) -> int:
        count = 0
        for pattern in patterns:
            if word.find(pattern) >= 0: count += 1

        return count
